source/numerics/data_out_dof_data.cc

```diff
@@ -527,6 +527,45 @@ namespace internal
         return vector[cell_number];
       }
 
+      // FIXME:
+      // explicit instantiations for PETSc complex vectors
+#ifdef PETSC_USE_COMPLEX
+      template <>
```

Comment:
You don't need the `template <>` here, see the existing specialization below. (In fact, it's not a specialization but simply an overload.)